pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -12,167 +12,157 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// AddOrRefcountVolumeConfig makes sure we have a VolumeConfig with a non-zero
-// RefCount
-func AddOrRefcountVolumeConfig(ctx *zedmanagerContext, blobSha256 string,
-	appInstID uuid.UUID, volumeID uuid.UUID, purgeCounter uint32, ss types.StorageStatus) {
-
-	key := types.VolumeKeyFromParts(blobSha256, appInstID, volumeID, purgeCounter)
-	log.Infof("AddOrRefcountVolumeConfig for %s", key)
-	m := lookupVolumeConfig(ctx, key)
+// MaybeAddVolumeRefConfig publishes volume ref config with refcount
+// to the volumemgr
+func MaybeAddVolumeRefConfig(ctx *zedmanagerContext, appInstID uuid.UUID,
+	vrc types.VolumeRefConfig) {
```

Comment:
Things are easier to understand if the VolumeRefConfig is created by this function, and you pass in the volumeID and generationCounter. That way all the refcount logic belongs in this file.

Reply:
Okay, I will change it. Thanks!